src/protocol/attribute.rs

```diff
@@ -0,0 +1,224 @@
+use super::{AttributionModel, RecordId};
+use crate::{
+    error::BoxError,
+    field::Field,
+    helpers::mesh::{Gateway, Mesh},
+    protocol::context::ProtocolContext,
+    secret_sharing::Replicated,
+};
+use futures::future::try_join_all;
+
+/// Obilivious attribution protocol
+///
+/// Obliviously attribute trigger events to corresponding source events for different breakdowns in a MPC.
+/// There are several common attribution models such as Last Touch, Even Credit and Exponential Decay.
+/// https://github.com/patcg-individual-drafts/ipa/blob/main/IPA-End-to-End.md#oblivious-last-touch-attribution
+#[allow(dead_code)]
+pub struct Attribute<'a, G> {
```

Comment:
"Attribute" is maybe not the best choice of name for this, given the potential for it to be confused.  "Attribution" is probably cleaner, even if it doesn't have the right tense.